lib/rucio/common/utils.py

```diff
@@ -1214,13 +1214,23 @@ def api_update_return_dict(dictionary):
 
     copied = False  # Avoid side effects from pass by object
 
-    if 'rse_id' in dictionary.keys():
-        if 'rse' not in dictionary.keys():
-            if not copied:
-                dictionary = dictionary.copy()
-                copied = True
-            import rucio.core.rse
-            dictionary['rse'] = rucio.core.rse.get_rse_name(rse_id=dictionary['rse_id'])
+    for rse_str in ['rse', 'src_rse', 'source_rse', 'dest_rse', 'destination_rse']:
+        rse_id_str = '%s_id' % rse_str
+        if rse_id_str in dictionary.keys() and dictionary[rse_id_str] is not None:
+            if rse_str not in dictionary.keys():
+                if not copied:
+                    dictionary = dictionary.copy()
+                    copied = True
+                import rucio.core.rse
+                dictionary[rse_str] = rucio.core.rse.get_rse_name(rse_id=dictionary[rse_id_str])
+
+    # if 'rse_id' in dictionary.keys():
```

Comment:
Is the old way of doing this left commented out in case the new method doesn't work out (like for an easier reversion)? Will this be removed entirely later if it does work out?

Reply:
That was my initial plan, yes (and again I forgot to remove it later). Since the new method seems to pass all the tests, I'll take this out.